tests/client_api.cpp

```diff
@@ -1593,61 +1593,73 @@ TEST(ClientAPI, PublicRooms)
         while (alice->access_token().empty() || bob->access_token().empty())
                 sleep();
 
+        std::cout << "Alice AT: " << alice->access_token() << "\n";
+        std::cout << "Bob AT: " << bob->access_token() << "\n";
+
         mtx::requests::CreateRoom req;
         req.name = "Public Room";
         req.topic = "Test";
         req.visibility = Visibility::Public;
         req.invite = {"@bob:localhost"};
+        // req.room_alias_name = "foo";
 
         alice->create_room(
                 req, [alice, bob](const mtx::responses::CreateRoom &res, RequestErr err){
                 check_error(err);
-                auto room_id = res.room_id.to_string();
-                bob->join_room(room_id, [](const mtx::responses::RoomId &, RequestErr err) {
-                        check_error(err);
-                });
+                auto room_id = res.room_id;
 
-                // TEST 1: endpoints to set and get the visibility of the room we just created
-                json body = {{"visibility", "public"}};
-                alice->put_room_visibility(room_id, body, [alice](RequestErr err){
+                bob->join_room(room_id.to_string(), [alice, bob, room_id](const mtx::responses::RoomId &, RequestErr err) {
                         check_error(err);
-                });
 
-                alice->get_room_visibility
-                (room_id, [alice](const mtx::responses::RoomVisibility &res, RequestErr err){
-                        check_error(err);
-                        EXPECT_EQ(visibilityToString(res.visibility), "public");
-                });
+                        // TEST 1: endpoints to set and get the visibility of the room we just created
+                        mtx::requests::RoomVisibility r;
+                        r.visibility = mtx::requests::Visibility::Private;
 
-                alice->get_room_visibility
-                ("", [alice](const mtx::responses::RoomVisibility &, RequestErr err) {
-                        ASSERT_TRUE(err);
-                        EXPECT_EQ(mtx::errors::to_string(err->matrix_error.errcode),
-                        "M_NOT_FOUND");
-                });
+                        std::cout << "HERE\n";
 
-                // TEST 2: endpoints to add and list the public rooms on the server 
-                mtx::requests::PublicRooms room_req;
-                room_req.limit = 1;
-                room_req.include_all_networks = true;
-                json j = room_req;
-                std::cout << j.dump(4) << "\n";
+                        alice->put_room_visibility(room_id.to_string(), r, [alice, bob, room_id](RequestErr err){
+                                check_error(err);
 
-                alice->post_public_rooms
-                (room_req, [alice](const mtx::responses::PublicRooms &, RequestErr err) {
-                        check_error(err);
-                });
+                                std::cout << "NOW IM HERE\n";
```

Comment:
@deepbluev7 This is where I'm currently stuck. The put wrapper above executes, and the in it the correct encoded path is printed, but the put request doesn't seem to be called right because this doesn't print.

Reply:
```
[ RUN      ] ClientAPI.PublicRooms
Alice AT: MDAxN2xvY2F0aW9uIGxvY2FsaG9zdAowMDEzaWRlbnRpZmllciBrZXkKMDAxMGNpZCBnZW4gPSAxCjAwMjNjaWQgdXNlcl9pZCA9IEBhbGljZTpsb2NhbGhvc3QKMDAxNmNpZCB0eXBlID0gYWNjZXNzCjAwMjFjaWQgbm9uY2UgPSBNLjBALCZCbFo5QVU4T24rCjAwMmZzaWduYXR1cmUgA5LzsAOLVwxZD6f_0jubxESsO3Cp_JVNa2l1ezzSnVkK
Bob AT: MDAxN2xvY2F0aW9uIGxvY2FsaG9zdAowMDEzaWRlbnRpZmllciBrZXkKMDAxMGNpZCBnZW4gPSAxCjAwMjFjaWQgdXNlcl9pZCA9IEBib2I6bG9jYWxob3N0CjAwMTZjaWQgdHlwZSA9IGFjY2VzcwowMDIxY2lkIG5vbmNlID0gKlpYbC0tTTVxQnorLmJaNgowMDJmc2lnbmF0dXJlIJ6riJb_UDP-b_rC1K-ZBi4MQ2GrZXJoR4XFNnNVXjpUCg
HERE
/client/r0/directory/list/room/%21nTDMyBIvhMFafRsloT%3Alocalhost
[       OK ] ClientAPI.PublicRooms (3769 ms)
```

Works fine for me

Reply:
How so? My very point is, after the put call in `put_room_visibility`, the subsequent GET request(s) (or print statement of "NOW IM HERE", in this case) in its callback doesn't execute.